### PR TITLE
[JW8-11938] Sanitize title and description

### PR DESCRIPTION
**Pull request by zetagame**
### This PR will...
Sanitize title and description fields.

### Why is this Pull Request needed?
Patches an XSS vulnerability.

#### Addresses Issue(s):

JW8-11938

### Checklist
- [ ] Jenkins builds and unit tests are passing
- [ ] I have reviewed the automated results

### CODE REVIEW

**Review by Dawolee — APPROVED** [review]

File: src/js/utils/dom.js [hunk 1]
```diff
@@ -279,3 +279,10 @@ export function deviceIsLandscape() {
 
     return isLandscape || (window.orientation === 90 || window.orientation === -90);
 }
+
+// Removes all unallowed HTML tags/attributes from a template string. Useful for user created HTML.
+export function HTMLSafeString(str) {
+    return str.replaceAll(/&|<|>|"|''/gm, function(i) {
+        return '&#'+i.charCodeAt(0)+';';
+     }).replaceAll(/&#60;(\/?)(b|strong|i|em|p|br|ul|ol|li|h.)&#62;/gmi, '<$1$2>');
```
Comment (nebnitt) [inline src/js/utils/dom.js:287]: Curious where this regex came from and this overall func? Home-grown or perhaps pulled from SO or elsewhere?

File: src/js/view/title.ts [hunk 2]
```diff
@@ -74,10 +75,10 @@ class Title {
             let title = '';
             let description = '';
 
-            if (item.title && model.get('displaytitle')) {
+            if (typeof item.title === 'string' && model.get('displaytitle')) {
```
Comment (nebnitt) [inline src/js/view/title.ts:78]: Do we definitely need these `typeof` checks here? I'm not sure the implications of adding these and whether it'll result in `title` or `description` being blank in certain cases. Wondering if we can add the `typeof` checks into `HTMLSafeString`. Or maybe it's by design that we want to disallow any non-string values here?

File: src/js/utils/dom.js (hunk 1 above)
Comment (zetagame) [reply src/js/utils/dom.js:287]: homegrown. Can break it down if you want

File: src/js/view/title.ts (hunk 2 above)
Comment (zetagame) [reply src/js/view/title.ts:78]: Yep. PlayListItem isn't strongly typed (it's imported from JS)

**Pull request merged**